server/routes/api/disease.py

```diff
@@ -32,3 +37,47 @@ def get_node(dcid):
                                    post=False,
                                    has_payload=False)
   return response
+
+
+@bp.route('/diseaseParent/<path:dcid>')
+def get_disease_parent(dcid):
+  """Returns parent node for given a disease node."""
+  # list to store parent node
+  list_parent = []
+  # disease of final parent
+  DISEASE_DCID = "bio/DOID_4"
+  curr_dcid = dcid
+  # dcid of the biggest parent node where iteration stops
+  while (curr_dcid != DISEASE_DCID):
+    node_dcids = dc_service.property_values([curr_dcid],
+                                            "specializationOf").get(
+                                                curr_dcid, [])
+    if not node_dcids:
+      break
+    node_dcid = node_dcids[0]
+    node_names = dc_service.property_values([node_dcid],
+                                            "name").get(node_dcid, [])
+    node_name = node_dcid
+    if node_names:
+      node_name = node_names[0]
+    list_parent.append(DiseaseParent(node_dcid, node_name))
+    curr_dcid = node_dcid
+  # return a list of dcid and name lists
+  return Response(json.dumps(list_parent, cls=DiseaseParentEncoder),
+                  200,
+                  mimetype='application/json')
+
+
+# class which defines an object storing parent dcid and name
+class DiseaseParent:
```

Comment:
I think you can just define a dataclass like this: https://github.com/datacommonsorg/website/blob/master/server/lib/nl/page_config_builder.py#L93

& you should define it before it is used

Reply:
Thank you! Using dataclass is much easier!